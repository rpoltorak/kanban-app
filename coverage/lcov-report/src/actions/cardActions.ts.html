<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/actions/cardActions.ts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/actions</a> cardActions.ts
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>11/11</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { Action } from "redux";
import uuid from "uuid";
&nbsp;
import CardModel from "../models/cardModel";
&nbsp;
export enum CardActionTypes {
  CARD_CREATE = "CARD_CREATE",
  CARD_DELETE = "CARD_DELETE",
  CARD_UPDATE = "CARD_UPDATE",
}
&nbsp;
export type CardAction = CreateCardAction | DeleteCardAction | UpdateCardAction;
&nbsp;
export interface CreateCardAction extends Action {
  type: CardActionTypes.CARD_CREATE;
  payload: {
    card: CardModel;
  };
}
&nbsp;
export interface DeleteCardAction extends Action {
  type: CardActionTypes.CARD_DELETE;
  payload: {
    cardId: string;
    laneId: string;
  };
}
&nbsp;
export interface UpdateCardAction extends Action {
  type: CardActionTypes.CARD_UPDATE;
  payload: {
    cardId: string;
    changes: Partial&lt;CardModel&gt;;
  };
}
&nbsp;
export function createCard(title: string, description: string, laneId: string): CreateCardAction {
  return {
    type: CardActionTypes.CARD_CREATE,
    payload: {
      card: {
        id: uuid.v4(),
        laneId,
        title,
        description,
      },
    },
  };
}
&nbsp;
export function deleteCard(cardId: string, laneId: string): DeleteCardAction {
  return {
    type: CardActionTypes.CARD_DELETE,
    payload: {
      cardId,
      laneId,
    },
  };
}
&nbsp;
export function updateCard(cardId: string, changes: Partial&lt;CardModel&gt;): UpdateCardAction {
  return {
    type: CardActionTypes.CARD_UPDATE,
    payload: {
      cardId,
      changes,
    },
  };
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 12 2018 23:33:21 GMT+0100 (Central European Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
