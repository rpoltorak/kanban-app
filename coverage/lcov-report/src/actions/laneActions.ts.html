<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/actions/laneActions.ts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/actions</a> laneActions.ts
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>11/11</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { Action } from "redux";
import uuid from "uuid";
&nbsp;
import LaneModel from "../models/laneModel";
&nbsp;
export enum LaneActionTypes {
  LANE_CREATE = "LANE_CREATE",
  LANE_DELETE = "LANE_DELETE",
  LANE_UPDATE = "LANE_UPDATE",
}
&nbsp;
export type LaneAction = CreateLaneAction | DeleteLaneAction | UpdateLaneAction;
&nbsp;
export interface CreateLaneAction extends Action {
  type: LaneActionTypes.LANE_CREATE;
  payload: {
    lane: LaneModel,
  };
}
&nbsp;
export interface DeleteLaneAction extends Action {
  type: LaneActionTypes.LANE_DELETE;
  payload: {
    laneId: string,
  };
}
&nbsp;
export interface UpdateLaneAction extends Action {
  type: LaneActionTypes.LANE_UPDATE;
  payload: {
    laneId: string;
    changes: Partial&lt;LaneModel&gt;,
  };
}
&nbsp;
export function createLane(title: string): CreateLaneAction {
  return {
    type: LaneActionTypes.LANE_CREATE,
    payload: {
      lane: {
        id: uuid.v4(),
        title,
        cards: [],
      },
    },
  };
}
&nbsp;
export function deleteLane(laneId: string): DeleteLaneAction {
  return {
    type: LaneActionTypes.LANE_DELETE,
    payload: {
      laneId,
    },
  };
}
&nbsp;
export function updateLane(laneId: string, changes: Partial&lt;LaneModel&gt;): UpdateLaneAction {
  return {
    type: LaneActionTypes.LANE_UPDATE,
    payload: {
      laneId,
      changes,
    },
  };
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 12 2018 23:33:21 GMT+0100 (Central European Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
