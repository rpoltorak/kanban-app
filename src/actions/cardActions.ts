import { Action } from "redux";
import uuid from "uuid";

import CardModel from "../models/cardModel";

export enum CardActionTypes {
  CARD_CREATE = "CARD_CREATE",
  CARD_DELETE = "CARD_DELETE",
  CARD_UPDATE = "CARD_UPDATE",
}

export type CardAction = CreateCardAction | DeleteCardAction | UpdateCardAction;

export interface CreateCardAction extends Action {
  type: CardActionTypes.CARD_CREATE;
  payload: {
    card: CardModel;
  };
}

export interface DeleteCardAction extends Action {
  type: CardActionTypes.CARD_DELETE;
  payload: {
    cardId: string;
    laneId: string;
  };
}

export interface UpdateCardAction extends Action {
  type: CardActionTypes.CARD_UPDATE;
  payload: {
    cardId: string;
    changes: Partial<CardModel>;
  };
}

export function createCard(title: string, description: string, laneId: string): CreateCardAction {
  return {
    type: CardActionTypes.CARD_CREATE,
    payload: {
      card: {
        id: uuid.v4(),
        laneId,
        title,
        description,
      },
    },
  };
}

export function deleteCard(cardId: string, laneId: string): DeleteCardAction {
  return {
    type: CardActionTypes.CARD_DELETE,
    payload: {
      cardId,
      laneId,
    },
  };
}

export function updateCard(cardId: string, changes: Partial<CardModel>): UpdateCardAction {
  return {
    type: CardActionTypes.CARD_UPDATE,
    payload: {
      cardId,
      changes,
    },
  };
}
