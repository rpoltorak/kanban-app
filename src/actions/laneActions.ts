import { Action } from "redux";
import uuid from "uuid";

import LaneModel from "../models/laneModel";

export enum LaneActionTypes {
  LANE_CREATE = "LANE_CREATE",
  LANE_DELETE = "LANE_DELETE",
  LANE_UPDATE = "LANE_UPDATE",
}

export type LaneAction = CreateLaneAction | DeleteLaneAction | UpdateLaneAction;

export interface CreateLaneAction extends Action {
  type: LaneActionTypes.LANE_CREATE;
  payload: {
    lane: LaneModel,
  };
}

export interface DeleteLaneAction extends Action {
  type: LaneActionTypes.LANE_DELETE;
  payload: {
    laneId: string,
  };
}

export interface UpdateLaneAction extends Action {
  type: LaneActionTypes.LANE_UPDATE;
  payload: {
    laneId: string;
    changes: Partial<LaneModel>,
  };
}

export function createLane(title: string): CreateLaneAction {
  return {
    type: LaneActionTypes.LANE_CREATE,
    payload: {
      lane: {
        id: uuid.v4(),
        title,
        cards: [],
      },
    },
  };
}

export function deleteLane(laneId: string): DeleteLaneAction {
  return {
    type: LaneActionTypes.LANE_DELETE,
    payload: {
      laneId,
    },
  };
}

export function updateLane(laneId: string, changes: Partial<LaneModel>): UpdateLaneAction {
  return {
    type: LaneActionTypes.LANE_UPDATE,
    payload: {
      laneId,
      changes,
    },
  };
}
