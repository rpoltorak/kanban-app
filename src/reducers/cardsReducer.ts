import { LaneActionTypes } from "../actions/laneActions";
import { CardActionTypes } from "../actions/cardActions";
import { RootAction } from "./rootReducer";
import CardModel from "../models/cardModel";

export interface CardsState {
  byId: {
    [cardId: string]: CardModel;
  };
  allIds: string[];
}

const initialState = {
  byId: {},
  allIds: [],
};

export default (state: CardsState = initialState, action: RootAction) => {
  switch (action.type) {
    case CardActionTypes.CARD_CREATE:
      return createCard(state, action.payload.card);

    case CardActionTypes.CARD_DELETE:
      return deleteCard(state, action.payload.cardId);

    case CardActionTypes.CARD_UPDATE: {
      const { cardId, changes } = action.payload;
      return updateCard(state, cardId, changes);
    }

    case LaneActionTypes.LANE_DELETE:
      return deleteLane(state, action.payload.laneId);

    default: return state;
  }
};

function createCard(state: CardsState, card: CardModel): CardsState {
  return {
    byId: {
      ...state.byId,
      [card.id]: card,
    },
    allIds: [...state.allIds, card.id],
  };
}

function deleteCard(state: CardsState, cardId: string): CardsState {
  const {[cardId]: deletedCard, ...otherCards} = state.byId;

  return {
    byId: otherCards,
    allIds: state.allIds.filter((id: string): boolean => id !== cardId),
  };
}

function updateCard(state: CardsState, cardId: string, changes: Partial<CardModel>): CardsState {
  const card = state.byId[cardId];

  return {
    allIds: state.allIds,
    byId: {
      ...state.byId,
      [cardId]: {
        ...card,
        ...changes,
      },
    },
  };
}

function deleteLane(state: CardsState, laneId: string): CardsState {
  const cards = Object.values(state.byId).filter((card: CardModel) => card.laneId !== laneId);

  return {
    byId: cards.reduce((byId: { [cardId: string]: CardModel }, card: CardModel) => {
      byId[card.id] = card;

      return byId;
    }, {}),
    allIds: cards.map((card: CardModel) => card.id),
  };
}
