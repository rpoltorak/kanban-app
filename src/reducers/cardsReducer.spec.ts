
import cardsReducer from "./cardsReducer";
import CardModel from "../models/cardModel";
import { createCard, deleteCard, updateCard } from "../actions/cardActions";
import { deleteLane } from "../actions/laneActions";

const card: CardModel = { id: "21", laneId: "1", title: "current title", description: "current description"  };
const otherCard: CardModel = { id: "22", laneId: "1", title: "current title", description: "current description"  };
const updatedCard: CardModel = { id: "21", laneId: "1", title: "updated title", description: "updated description" };

const initialState = cardsReducer(undefined, {} as any);

describe("Cards Reducer", () => {

  describe("snapshot", () => {
    it("should match a snapshot", () => {
      expect(initialState).toMatchSnapshot();
    });
  });

  describe("card_create", () => {
    it("should add a new card as the first element", () => {
      const action = createCard("title", "description", "1");
      const state = cardsReducer(initialState, action);
      const { id } = action.payload.card;

      expect(state.byId[id]).toEqual(action.payload.card);
      expect(state.allIds).toHaveLength(1);
      expect(state.allIds[0]).toEqual(id);
    });
  });

  describe("card_delete", () => {
    it("should delete existing card with specified id", () => {
      const prevState = {
        ...initialState,
        byId: {
          [card.id]: card,
        },
        allIds: [card.id],
      };

      const action = deleteCard(card.id, card.laneId);
      const nextState = cardsReducer(prevState, action);

      expect(nextState.allIds).toHaveLength(0);
      expect(nextState.byId[card.id]).toBeUndefined();
    });
  });

  describe("card_update", () => {
    it("should update existing card's title and description", () => {
      const prevState = {
        ...initialState,
        byId: {
          [card.id]: card,
        },
        allIds: [card.id],
      };

      const action = updateCard(card.id, { title: "updated title", description: "updated description" });
      const nextState = cardsReducer(prevState, action);

      expect(nextState.byId[card.id]).toEqual(updatedCard);
    });
  });

  describe("lane_delete", () => {
    it("should delete all cards related to deleted lane", () => {
      const prevState = {
        ...initialState,
        byId: {
          [card.id]: card,
          [otherCard.id]: otherCard,
        },
        allIds: [card.id, otherCard.id],
      };

      const action = deleteLane("1");
      const nextState = cardsReducer(prevState, action);

      expect(nextState.byId).toEqual({});
      expect(nextState.allIds).toHaveLength(0);
    });
  });
});
