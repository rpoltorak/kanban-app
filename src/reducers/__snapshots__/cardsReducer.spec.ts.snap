// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards Reducer snapshot should match a snapshot 1`] = `
Object {
  "allIds": Array [],
  "byId": Object {},
}
`;
